### ISSUE TITLE
TaskDefinition.Options generated without field

### ISSUE THREAD

**Issue by ndeloof** [awslabs/goformation#297]
after code has been regenerated on https://github.com/awslabs/goformation/pull/296

TaskDefinition_LogConfiguration.Options, which was previously a `map[string]string`, now is implemented as a `*TaskDefinition_Options`, but the latter has no field that can be used to pass options

**Comment by ndeloof** [comment]
I wonder this is an issue with the schema, as  "AWS::ECS::TaskDefinition.Options" is declared with `additionalProperties: false`
Was previously: 
```json
"Options": {
                    "additionalProperties": true,
                    "patternProperties": {
                        "^[a-zA-Z0-9]+$": {
                            "type": "string"
                        }
                    },
                    "type": "object"
                }
```

**Comment by ndeloof** [comment]
Looking at cloudformation schema https://d1uauaxba7bl26.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json I can't find the definition for the `Options` type.

**Comment by jonahjon** [comment]
> Looking at cloudformation schema https://d1uauaxba7bl26.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json I can't find the definition for the `Options` type.

Where did you get this link from? When I dig into the link https://d1uauaxba7bl26.cloudfront.net/ it looks like the most recent version is from 2017, if the XML "<LastModified></LastModified>" tag is correct.

**Comment by ndeloof** [comment]
This is the one used by `generate` : https://github.com/awslabs/goformation/blob/master/generate/main.go#L13

**Comment by jonahjon** [comment]
> This is the one used by `generate` : https://github.com/awslabs/goformation/blob/master/generate/main.go#L13

Ok, cool makes sense

**Comment by PaulMaddox** [comment]
The CloudFormation schema has the following, which looks correct to me at first glance. 

This looks correct, and should be generating a `map[string]string` in goformation. Let me do some further investigation to work out what is going wrong here...

```
"AWS::ECS::TaskDefinition.LogConfiguration": {
    "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions-logconfiguration.html",
    "Properties": {
    "LogDriver": {
        "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions-logconfiguration.html#cfn-ecs-taskdefinition-containerdefinition-logconfiguration-logdriver",
        "UpdateType": "Immutable",
        "Required": true,
        "PrimitiveType": "String"
    },
    "Options": {
        "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions-logconfiguration.html#cfn-ecs-taskdefinition-containerdefinition-logconfiguration-options",
        "UpdateType": "Immutable",
        "Required": false,
        "Type": "Map",
        "PrimitiveItemType": "String"
    },
    "SecretOptions": {
        "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions-logconfiguration.html#cfn-ecs-taskdefinition-logconfiguration-secretoptions",
        "UpdateType": "Immutable",
        "Required": false,
        "Type": "List",
        "ItemType": "Secret"
    }
    }
},
```

**Comment by PaulMaddox** [comment]
It looks like this was actually corrected last night in the upstream CloudFormation specification, and the goformation code is generating properly now. 

Unfortunately there is another CloudFormation schema issue introduced that is causing our tests to fail and blocking a release. See issue #300. I have raised this internally. 

![screenshot](https://user-images.githubusercontent.com/4678446/91121233-6a44a800-e6a8-11ea-885b-2930d1c98c3b.png)

**Comment by PaulMaddox** [comment]
Fixed in #320
